lib/action_controller/serialization.rb

```diff
@@ -25,6 +25,11 @@ def get_serializer(resource, options = {})
           "Please pass 'adapter: false' or see ActiveSupport::SerializableResource#serialize"
         options[:adapter] = false
       end
+      if resource.respond_to?(:current_page) && resource.respond_to?(:total_pages)
+        options[:pagination] = {}
+        options[:pagination][:original_url] = original_url
+        options[:pagination][:query_parameters] = query_parameters
+      end
```

Comment:
@bacarini Yeah, the idea is that since we only need to get the current url etc from the context when paginating, instead of manipulating the request object in the controller, and pass in options, we can pass in the `context` (that is, the request object) and only use it when paginating.  The details of it being an `option` param I don't totally like, but it is a pattern used in Rails to have an evaluation context, which I do like a bit better for letting moving the responsibility of understanding the context from the controller to the Link Paginator.  Also, it adds less methods to the controller.

That make sense?